assets/js/components/product-category-control/index.js

```diff
@@ -2,40 +2,24 @@
  * External dependencies
  */
 import { __, _n, sprintf } from '@wordpress/i18n';
-import { addQueryArgs } from '@wordpress/url';
-import apiFetch from '@wordpress/api-fetch';
 import { Component, Fragment } from '@wordpress/element';
 import { find } from 'lodash';
 import PropTypes from 'prop-types';
 import { SearchListControl, SearchListItem } from '@woocommerce/components';
 import { SelectControl } from '@wordpress/components';
-import { ENDPOINTS } from '@woocommerce/block-settings';
 
 /**
  * Internal dependencies
  */
+import { withCategories } from '../../hocs';
+import ErrorMessage from '../api-error-placeholder/error-message.js';
 import './style.scss';
 
 class ProductCategoryControl extends Component {
 	constructor() {
 		super( ...arguments );
-		this.state = {
-			list: [],
-			loading: true,
-		};
-		this.renderItem = this.renderItem.bind( this );
-	}
 
-	componentDidMount() {
-		apiFetch( {
-			path: addQueryArgs( `${ ENDPOINTS.products }/categories`, { per_page: -1 } ),
-		} )
-			.then( ( list ) => {
-				this.setState( { list, loading: false } );
-			} )
-			.catch( () => {
-				this.setState( { list: [], loading: false } );
-			} );
+		this.renderItem = this.renderItem.bind( this );
```

Comment:
Same here as what I commented on the `product-attribute-control`... this seems like a good candidate for converting to a functional component too now.

Reply:
Done in d4e2ddd9cefaa32d9565fe104c357e087ebeed8f.